src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -2,19 +2,109 @@ package cromwell.binding
 
 import java.io.File
 
-import cromwell.binding.WdlNamespace.ImportResolver
-import cromwell.binding.command._
 import cromwell.binding.types._
 import cromwell.binding.values.WdlValue
 import cromwell.parser.{AstTools, WdlParser}
 import cromwell.parser.WdlParser._
 import cromwell.util.FileUtil
-import AstTools.EnhancedAstNode
+import cromwell.parser.AstTools.{AstNodeName, EnhancedAstNode, EnhancedAstSeq}
 import scala.collection.JavaConverters._
 import scala.util.{Failure, Success, Try}
 import scala.language.postfixOps
 
 /**
+ * Define WdlNamespace as a sum type w/ two states - one containing a local workflow and one without.
+ * The latter is a valid state for a WDL file, however only the former can be requested to be run, so
+ * any constructs (e.g. WorkflowManagerActor) expecting to run a workflow should only take the `NamespaceWithWorkflow`
+ */
+sealed trait WdlNamespace extends WdlValue {
+  final val wdlType = WdlNamespaceType
+
+  def ast: Ast // FIXME: I think this is only used by the syntax highlighting, can it go away once we're built?
+  def importedAs: Option[String] // Used when imported with `as` 
+  def imports: Seq[Import] // FIXME: Change to Set?
+  def namespaces: Seq[WdlNamespace] // FIXME: Change to Set? FIXME: Rename to importedNamespaces?
+  def tasks: Seq[Task] // FIXME: Change to Set?
+  def importedWorkflows: Seq[Workflow] // FIXME: Change to Set?
+  def terminalMap: Map[Terminal, WdlSource]
+  def allWorkflows: Set[Workflow]
+
+  // Convenience method for findTask in the context of this namespace
+  def findTask(name: String): Option[Task] = WdlNamespace.findTask(name, namespaces, tasks)
+
+  override def toRawString = ???
+}
+
+/**
+ * A valid Namespace which doesn't have a locally defined Workflow. This should pass any validity checking but is not
+ * directly runnable by `WorkflowManagerActor`
+ */
```

Comment:
It seems to me like it'd be a lot less code to not have these two classes and instead just allow WdlNamespace to have 0+ workflows and then in the workflow runner just do a simple check to make sure that the submitted workflow has 1 workflow.

This could also set things up better in the future because we talked about having many workflows defined but only one defined as public.